### PR TITLE
Add a check for dead links and markdown linting

### PR DESCRIPTION
**Pull request by mark-pkgflow**
## Summary

This PR introduces quality checks and formatting standards for Markdown files in our project. It adds linting and formatting capabilities via Prettier, along with link validation using `html-proofer`. The goal is to ensure consistent formatting in our Markdown content and to verify the integrity of links in our compiled output.

## Changes

- **Linting and Formatting with Prettier**: Added Prettier as a development dependency to enforce consistent styling in our Markdown files. Configuration settings for Prettier are defined in `package.json`.
- **Link Validation with html-proofer**: Integrated `html-proofer` to check for broken links, images, and scripts in the compiled site output, ensuring that our content is reliable and web-ready.
- **Makefile Commands**: Introduced new commands in the `Makefile` to easily run linting, formatting, and link checks as part of our development workflow.

### Commits

1. **Add `html-proofer`, `prettier` and Config**:
  - Integrated `html-proofer` into the project to validate the integrity of links, images, and scripts in the compiled site output.
  - Added Prettier for Markdown linting and formatting, along with the necessary configuration in `package.json`.
2. **MD Lint Formatting**: Applied Prettier formatting to existing Markdown files to ensure consistency across our content.
3. **Fixed Links**: Addressed and fixed any broken or placeholder links identified by `html-proofer`, improving the reliability of our content.
4. **Documentation**: Added a document to help contributors understand and fix issues. ([view](https://github.com/DFE-Digital/improve-school-workload-and-wellbeing/blob/e62ec612c143a017ff1eadae2d6620ef83277a82/docs/resolving-link-and-linting-issues.md))

## Usage

- To check Markdown formatting: `make lint-md`
- To format Markdown files: `make format-md`
- To validate links post-build: `make check-links`

These commands enhance our development process by ensuring content quality and consistency.

## Dependencies Added

- **Prettier**: For linting and formatting Markdown files.
- **html-proofer**: For validating links, images, and scripts in the site's output.

### Note on `markdownlint`

We also evaluated the `markdownlint` gem as a potential tool for enforcing Markdown best practices. While `markdownlint` provides comprehensive linting capabilities and is highly configurable, we ultimately decided against its inclusion in our current setup. The primary reason was its lack of an auto-format feature, which we deemed essential for streamlining our content editing workflow. We sought a solution that not only identifies issues but also provides automatic corrections to ensure consistency and efficiency. As such, we opted for Prettier, which meets our requirements for both linting and auto-formatting of Markdown content.

## Continuous Integration and Deployment Enhancements

As part of this PR, we've also integrated the new linting and link validation checks into our continuous integration and deployment pipeline using GitHub Actions. This ensures that every push and pull request is automatically checked for Markdown formatting issues and broken links before merging, maintaining the high quality of our project's content and preventing potential issues in our live site.

### GitHub Actions Workflow

- ~**Markdown Linting Check**: A GitHub Action workflow has been added to run `make lint-md` on every push and pull request, ensuring that all Markdown files adhere to our formatting standards.~
- **Link Validation Check**: After the site is built, another GitHub Action workflow runs `make check-links` to validate all internal and external links, catching any broken links before they make it to production.

image

These automated checks are crucial for maintaining content quality and site reliability without slowing down our development process. They provide immediate feedback to contributors and prevent common content-related issues from affecting our live site.

---

Please review the changes and check that the changes have not altered the output HTML (ie anything on the preview app).

### CODE REVIEW

**Comment by mark-pkgflow** [comment]
[Check markdown and links on service](https://trello.com/c/fqgnNQJP/361-check-markdown-and-links-on-service)

File: docs/resolving-link-and-linting-issues.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+If the linting check fails, it typically means there are formatting
+discrepancies in your Markdown files.
+
+- **Identify Issues**: Review the output of the GitHub Actions run to see which
+  files and specific formatting rules were violated.
+- **Auto-Format Locally**: Run the following command in your local environment
+  to automatically format all Markdown files according to our project's
+  standards:
+
+  `bash make format-md`
```
Comment (AbigailMcP) [inline docs/resolving-link-and-linting-issues.md:33]: I'm just wondering how easy this is going to be for non-devs adding new resources via the GitHub UI. Does the linter fail a PR for line length, for example?

File: docs/resolving-link-and-linting-issues.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+A failed link validation check indicates there are broken or invalid links in
+your content.
+
+- **Identify Broken Links**: Examine the GitHub Actions output to find which
+  links failed the validation check and in which files.
+- **Manual Correction**: Update your Markdown files to fix or replace the broken
+  links. Use valid URLs and ensure placeholders like `TBC` or `example.com` are
+  replaced with final links.
+- **Test Locally**: After making changes, you can build the site locally and run
```
Comment (claire-hughez) [inline docs/resolving-link-and-linting-issues.md:48]: I'm not sure we can expect non-devs to be able to test locally. They might use the Github UI.

Comment (mark-pkgflow) [reply docs/resolving-link-and-linting-issues.md:48]: Agreed. we are removing any lint checking from the GitHub Actions

**Review by AbigailMcP — APPROVED** [review]

**Pull request merged**